can't derive Configure: wrong number of type arguments

```text
error[E0244]: wrong number of type arguments: expected 1, found 2
 --> src\config.rs:8:30
  |
8 | #[derive(Debug, Deserialize, Configure)]
  |                              ^^^^^^^^^ expected 1 type argument
```

any ideas?

ah ha, duh:

```
error[E0277]: the trait bound `serde::Deserialize<'_>` is not satisfied
```

so, i guess this is a bad error message. not sure you can do anything about that though.

expanded output:

<details>
<code><pre>
   #[allow(non_upper_case_globals, unused_attributes, unused_qualifications)]
    const _IMPL_DESERIALIZE_FOR_Config: () =
        {
            extern crate serde as _serde;
            #[automatically_derived]
            impl <'de> _serde::Deserialize<'de> for Config where
             Config: _serde::export::Default {
                fn deserialize<__D>(__deserializer: __D)
                 -> _serde::export::Result<Self, __D::Error> where
                 __D: _serde::Deserializer<'de> {
                    #[allow(non_camel_case_types)]
                    enum __Field {
                        __field0,
                        __field1,
                        __field2,
                        __field3,
                        __ignore,
                    }
                    struct __FieldVisitor;
                    impl <'de> _serde::de::Visitor<'de> for __FieldVisitor {
                        type
                        Value
                        =
                        __Field;
                        fn expecting(&self,
                                     formatter:
                                         &mut _serde::export::Formatter)
                         -> _serde::export::fmt::Result {
                            _serde::export::Formatter::write_str(formatter,
                                                                 "field identifier")
                        }
                        fn visit_u64<__E>(self, __value: u64)
                         -> _serde::export::Result<Self::Value, __E> where
                         __E: _serde::de::Error {
                            match __value {
                                0u64 => _serde::export::Ok(__Field::__field0),
                                1u64 => _serde::export::Ok(__Field::__field1),
                                2u64 => _serde::export::Ok(__Field::__field2),
                                3u64 => _serde::export::Ok(__Field::__field3),
                                _ =>
                                _serde::export::Err(_serde::de::Error::invalid_value(_serde::de::Unexpected::Unsigned(__value),
                                                                                     &"field index 0 <= i < 4")),
                            }
                        }
                        fn visit_str<__E>(self, __value: &str)
                         -> _serde::export::Result<Self::Value, __E> where
                         __E: _serde::de::Error {
                            match __value {
                                "ui" => _serde::export::Ok(__Field::__field0),
                                "manifest_path" =>
                                _serde::export::Ok(__Field::__field1),
                                "output_path" =>
                                _serde::export::Ok(__Field::__field2),
                                "host" =>
                                _serde::export::Ok(__Field::__field3),
                                _ => _serde::export::Ok(__Field::__ignore),
                            }
                        }
                        fn visit_bytes<__E>(self, __value: &[u8])
                         -> _serde::export::Result<Self::Value, __E> where
                         __E: _serde::de::Error {
                            match __value {
                                b"ui" =>
                                _serde::export::Ok(__Field::__field0),
                                b"manifest_path" =>
                                _serde::export::Ok(__Field::__field1),
                                b"output_path" =>
                                _serde::export::Ok(__Field::__field2),
                                b"host" =>
                                _serde::export::Ok(__Field::__field3),
                                _ => { _serde::export::Ok(__Field::__ignore) }
                            }
                        }
                    }
                    impl <'de> _serde::Deserialize<'de> for __Field {
                        #[inline]
                        fn deserialize<__D>(__deserializer: __D)
                         -> _serde::export::Result<Self, __D::Error> where
                         __D: _serde::Deserializer<'de> {
                            _serde::Deserializer::deserialize_identifier(__deserializer,
                                                                         __FieldVisitor)
                        }
                    }
                    struct __Visitor<'de> where
                           Config: _serde::export::Default {
                        marker: _serde::export::PhantomData<Config>,
                        lifetime: _serde::export::PhantomData<&'de ()>,
                    }
                    impl <'de> _serde::de::Visitor<'de> for __Visitor<'de>
                     where Config: _serde::export::Default {
                        type
                        Value
                        =
                        Config;
                        fn expecting(&self,
                                     formatter:
                                         &mut _serde::export::Formatter)
                         -> _serde::export::fmt::Result {
                            _serde::export::Formatter::write_str(formatter,
                                                                 "struct Config")
                        }
                        #[inline]
                        fn visit_seq<__A>(self, mut __seq: __A)
                         -> _serde::export::Result<Self::Value, __A::Error>
                         where __A: _serde::de::SeqAccess<'de> {
                            let __default: Self::Value =
                                _serde::export::Default::default();
                            let __field0 =
                                match match _serde::de::SeqAccess::next_element::<Ui>(&mut __seq)
                                          {
                                          ::result::Result::Ok(val) => val,
                                          ::result::Result::Err(err) => {
                                              return ::result::Result::Err(::convert::From::from(err))
                                          }
                                      } {
                                    _serde::export::Some(__value) => __value,
                                    _serde::export::None => {
                                        return _serde::export::Err(_serde::de::Error::invalid_length(0usize,
                                                                                                     &"tuple of 4 elements"));
                                    }
                                };
                            let __field1 =
                                match match _serde::de::SeqAccess::next_element::<PathBuf>(&mut __seq)
                                          {
                                          ::result::Result::Ok(val) => val,
                                          ::result::Result::Err(err) => {
                                              return ::result::Result::Err(::convert::From::from(err))
                                          }
                                      } {
                                    _serde::export::Some(__value) => __value,
                                    _serde::export::None => {
                                        return _serde::export::Err(_serde::de::Error::invalid_length(1usize,
                                                                                                     &"tuple of 4 elements"));
                                    }
                                };
                            let __field2 =
                                match match _serde::de::SeqAccess::next_element::<Option<PathBuf>>(&mut __seq)
                                          {
                                          ::result::Result::Ok(val) => val,
                                          ::result::Result::Err(err) => {
                                              return ::result::Result::Err(::convert::From::from(err))
                                          }
                                      } {
                                    _serde::export::Some(__value) => __value,
                                    _serde::export::None => {
                                        return _serde::export::Err(_serde::de::Error::invalid_length(2usize,
                                                                                                     &"tuple of 4 elements"));
                                    }
                                };
                            let __field3 =
                                match match _serde::de::SeqAccess::next_element::<analysis::AnalysisHost>(&mut __seq)
                                          {
                                          ::result::Result::Ok(val) => val,
                                          ::result::Result::Err(err) => {
                                              return ::result::Result::Err(::convert::From::from(err))
                                          }
                                      } {
                                    _serde::export::Some(__value) => __value,
                                    _serde::export::None => {
                                        return _serde::export::Err(_serde::de::Error::invalid_length(3usize,
                                                                                                     &"tuple of 4 elements"));
                                    }
                                };
                            _serde::export::Ok(Config{ui: __field0,
                                                      manifest_path: __field1,
                                                      output_path: __field2,
                                                      host: __field3,})
                        }
                        #[inline]
                        fn visit_map<__A>(self, mut __map: __A)
                         -> _serde::export::Result<Self::Value, __A::Error>
                         where __A: _serde::de::MapAccess<'de> {
                            let mut __field0: _serde::export::Option<Ui> =
                                _serde::export::None;
                            let mut __field1:
                                    _serde::export::Option<PathBuf> =
                                _serde::export::None;
                            let mut __field2:
                                    _serde::export::Option<Option<PathBuf>> =
                                _serde::export::None;
                            let mut __field3:
                                    _serde::export::Option<analysis::AnalysisHost> =
                                _serde::export::None;
                            while let _serde::export::Some(__key) =
                                      match _serde::de::MapAccess::next_key::<__Field>(&mut __map)
                                          {
                                          ::result::Result::Ok(val) => val,
                                          ::result::Result::Err(err) => {
                                              return ::result::Result::Err(::convert::From::from(err))
                                          }
                                      } {
                                match __key {
                                    __Field::__field0 => {
                                        if _serde::export::Option::is_some(&__field0)
                                           {
                                            return _serde::export::Err(<__A::Error
                                                                           as
                                                                           _serde::de::Error>::duplicate_field("ui"));
                                        }
                                        __field0 =
                                            _serde::export::Some(match _serde::de::MapAccess::next_value::<Ui>(&mut __map)
                                                                     {
                                                                     ::result::Result::Ok(val)
                                                                     => val,
                                                                     ::result::Result::Err(err)
                                                                     => {
                                                                         return ::result::Result::Err(::convert::From::from(err))
                                                                     }
                                                                 });
                                    }
                                    __Field::__field1 => {
                                        if _serde::export::Option::is_some(&__field1)
                                           {
                                            return _serde::export::Err(<__A::Error
                                                                           as
                                                                           _serde::de::Error>::duplicate_field("manifest_path"));
                                        }
                                        __field1 =
                                            _serde::export::Some(match _serde::de::MapAccess::next_value::<PathBuf>(&mut __map)
                                                                     {
                                                                     ::result::Result::Ok(val)
                                                                     => val,
                                                                     ::result::Result::Err(err)
                                                                     => {
                                                                         return ::result::Result::Err(::convert::From::from(err))
                                                                     }
                                                                 });
                                    }
                                    __Field::__field2 => {
                                        if _serde::export::Option::is_some(&__field2)
                                           {
                                            return _serde::export::Err(<__A::Error
                                                                           as
                                                                           _serde::de::Error>::duplicate_field("output_path"));
                                        }
                                        __field2 =
                                            _serde::export::Some(match _serde::de::MapAccess::next_value::<Option<PathBuf>>(&mut __map)
                                                                     {
                                                                     ::result::Result::Ok(val)
                                                                     => val,
                                                                     ::result::Result::Err(err)
                                                                     => {
                                                                         return ::result::Result::Err(::convert::From::from(err))
                                                                     }
                                                                 });
                                    }
                                    __Field::__field3 => {
                                        if _serde::export::Option::is_some(&__field3)
                                           {
                                            return _serde::export::Err(<__A::Error
                                                                           as
                                                                           _serde::de::Error>::duplicate_field("host"));
                                        }
                                        __field3 =
                                            _serde::export::Some(match _serde::de::MapAccess::next_value::<analysis::AnalysisHost>(&mut __map)
                                                                     {
                                                                     ::result::Result::Ok(val)
                                                                     => val,
                                                                     ::result::Result::Err(err)
                                                                     => {
                                                                         return ::result::Result::Err(::convert::From::from(err))
                                                                     }
                                                                 });
                                    }
                                    _ => {
                                        let _ =
                                            match _serde::de::MapAccess::next_value::<_serde::de::IgnoredAny>(&mut __map)
                                                {
                                                ::result::Result::Ok(val) =>
                                                val,
                                                ::result::Result::Err(err) =>
                                                {
                                                    return ::result::Result::Err(::convert::From::from(err))
                                                }
                                            };
                                    }
                                }
                            }
                            let __default: Self::Value =
                                _serde::export::Default::default();
                            let __field0 =
                                match __field0 {
                                    _serde::export::Some(__field0) =>
                                    __field0,
                                    _serde::export::None => __default.ui,
                                };
                            let __field1 =
                                match __field1 {
                                    _serde::export::Some(__field1) =>
                                    __field1,
                                    _serde::export::None =>
                                    __default.manifest_path,
                                };
                            let __field2 =
                                match __field2 {
                                    _serde::export::Some(__field2) =>
                                    __field2,
                                    _serde::export::None =>
                                    __default.output_path,
                                };
                            let __field3 =
                                match __field3 {
                                    _serde::export::Some(__field3) =>
                                    __field3,
                                    _serde::export::None => __default.host,
                                };
                            _serde::export::Ok(Config{ui: __field0,
                                                      manifest_path: __field1,
                                                      output_path: __field2,
                                                      host: __field3,})
                        }
                    }
                    const FIELDS: &'static [&'static str] =
                        &["ui", "manifest_path", "output_path", "host"];
                    _serde::Deserializer::deserialize_struct(__deserializer,
                                                             "Config", FIELDS,
                                                             __Visitor{marker:
                                                                           _serde::export::PhantomData::<Config>,
                                                                       lifetime:
                                                                           _serde::export::PhantomData,})
                }
            }
        };
    impl ::configure::Configure for Config {
        fn generate() -> Result<Self, ::configure::DeserializeError> {
            let deserializer =
                ::configure::source::CONFIGURATION.get("doxidize");
            ::serde::Deserialize::deserialize(deserializer)
        }
    }
</pre></code>
</details>

--- a/configure_derive/src/lib.rs
+++ b/configure_derive/src/lib.rs
@@ -33,7 +33,7 @@ fn impl_configure(ast: DeriveInput) -> Tokens {
 
     quote!{
         impl #generics ::configure::Configure for #ty #generics {
-            fn generate() -> Result<Self, ::configure::DeserializeError> {
+            fn generate() -> ::std::result::Result<Self, ::configure::DeserializeError> {
                 let deserializer = ::configure::source::CONFIGURATION.get(#project);
                 ::serde::Deserialize::deserialize(deserializer)
             }

